Userland/Utilities/groupadd.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019-2020, Jesse Buhagiar <jooster669@gmail.com>
+ * Copyright (c) 2021, Brandon Pruitt  <brapru@pm.me>
+ * Copyright (c) 2021, Maxime Friess <M4x1me@pm.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Base64.h>
+#include <AK/Random.h>
+#include <AK/String.h>
+#include <LibCore/ArgsParser.h>
+#include <crypt.h>
+#include <ctype.h>
+#include <errno.h>
+#include <grp.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+constexpr gid_t GROUPS_GID = 100;
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio wpath rpath cpath chown", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    int gid = 0;
+    char const* groupname = nullptr;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_option(gid, "Group ID (gid) for the new group", "gid", 'g', "gid");
+    args_parser.add_positional_argument(groupname, "Name of the group (groupname)", "group");
+
+    args_parser.parse(argc, argv);
+
+    // Let's run a quick sanity check on groupname
+    if (strpbrk(groupname, "\\/!@#$%^&*()~+=`:\n")) {
+        warnln("invalid character in groupname, {}", groupname);
+        return 1;
+    }
+
+    // Disallow names starting with _ and -
+    if (groupname[0] == '_' || groupname[0] == '-' || !isalpha(groupname[0])) {
```

Comment:
Luckily `!isalpha(groupname[0])` is false for the NUL character (e.g. if the user specified an empty string. Should we make that more explicit though?

Reply:
Do you mean documenting the `isalpha` function (out of the scope of that pr) or adding a comment saying we refuse empty string ?

Reply:
No, I mean, maybe check whether it's empty explicitly - seeing how the behavior of `isalpha` might not be obvious here.